src/operator/tensor/matrix_op.cc

```diff
@@ -210,24 +272,18 @@ static void FlattenEx(const nnvm::NodeAttrs& attrs,
 #endif
 }
 
+#if MXNET_USE_MKLDNN == 1
```

Comment:
Seems this will change the original behavior when MKL-DNN is not enabled. `FlattenStorageType` is not defined then.

Reply:
This is followed new style with other mkldnn op, that is, defining InferStorageType function within MKLDNN  macro. Most other ops are refactored into this style by Luobao, I guess this one is missing.